source/s3_auto_ranged_get.c

```diff
@@ -279,152 +280,129 @@ static int s_s3_auto_ranged_get_prepare_request(
     return AWS_OP_ERR;
 }
 
-static int s_s3_auto_ranged_get_header_block_done(
```

Comment:
Instead of using a mixture of the HTTP header callback and the notify_destroyed function later on, we instead use a new-and-improved request_finished function.  (The request will have stored the headers if we specified the appropriate flag when we created it during the update function.)